crates/services/p2p/src/codecs/request_response.rs

```diff
@@ -0,0 +1,150 @@
+use std::io;
+
+use crate::request_response::{
+    messages::{
+        RequestMessage,
+        V1ResponseMessage,
+        V2ResponseMessage,
+    },
+    protocols::RequestResponseProtocol,
+};
+use async_trait::async_trait;
+use futures::{
+    AsyncRead,
+    AsyncReadExt,
+    AsyncWriteExt,
+};
+use libp2p::request_response;
+use strum::IntoEnumIterator as _;
+
+use super::{
+    Decode,
+    Encode,
+    Encoder,
+    RequestResponseProtocols,
+};
+
+#[derive(Debug, Clone)]
+pub struct RequestResponseMessageHandler<Codec> {
+    pub(crate) codec: Codec,
+    /// Used for `max_size` parameter when reading Response Message
+    /// Necessary in order to avoid DoS attacks
+    /// Currently the size mostly depends on the max size of the Block
+    pub(crate) max_response_size: usize,
```

Comment:
We seem to convert this one to a `u64` wherever used. How about defining it as such?

```suggestion
    pub(crate) max_response_size: u64,
```

Reply:
Same comment as above, I avoid functional changes in refactoring PRs. 

However in this case I have created a reference issue #2459, referenced it in the code (see b94c4519c2), and opened a PR for it (https://github.com/FuelLabs/fuel-core/pull/2460).

Reply:
Thanks, makes sense :+1: